fauna/models.py

```diff
@@ -16,43 +16,82 @@ def __hash__(self):
         hash(self.name)
 
 
-class DocumentReference:
-    collection: Module
-    ref_id: str
+class BaseReference:
+    _collection: Module
+    _id: str
```

Comment:
Storing both `id` and `name` as `str` seems slightly inefficient, as `id` will always be a 64-bit integer. It doesn't matter that much, but it might be more clear to users if `id` was stored as an integer instead.

If you're going to keep using a string for both, then can you add some docs to this field? It would be nice to know that this stores `id` or `name` depending on the type of reference.

Reply:
I'll clean this up.